source/fluid/map_frame.d

```diff
@@ -144,6 +146,14 @@ class MapFrame : Frame {
         updateSize();
     }
 
+    void addChild(Node node, Vector2 vector)
+    in ([vector.tupleof].any!isFinite, format!"Given %s isn't valid, values must be finite"(vector))
+    do {
+        children ~= node;
+        positions[node].coords = vector;
```

Comment:
This won't work, you'd need to assign `positions[node] = MapPosition(vector)`

Reply:
Really? But wouldn't that also reset the drop direction?

Reply:
I just took a look at the `MapPosition` definition. As far as I see, both would do exactly the same thing. However, I don't mind changing it, so I will.